plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -93,52 +92,24 @@ export function WriteItForMeButtonContainer() {
 		return instructions.join( '\n' );
 	};
 
-	const getApiError = ( status?: number ) => {
-		const errorMessagesByStatus: Record< number, string > = {
-			429: __(
-				'There have been too many requests. Please wait for a few minutes and try again.',
-				'woocommerce'
-			),
-			408: __(
-				'It seems the server is taking too long to respond. This is an experimental feature, so please try again later.',
-				'woocommerce'
-			),
-		};
-
-		if ( status && errorMessagesByStatus[ status ] ) {
-			return errorMessagesByStatus[ status ];
-		}
-
-		return __(
-			`Apologies, this is an experimental feature and there was an error with this service. Please try again.`,
-			'woocommerce'
+	if ( completionActive ) {
+		return (
+			<button
+				className="button wp-media-button woo-ai-write-it-for-me-btn"
+				type="button"
+				title={ __(
+					'Stop generating the description.',
+					'woocommerce'
+				) }
+				onClick={ () => {
+					stopCompletion();
+				} }
+			>
+				<img src={ MagicIcon } alt="" />
+				{ __( 'Stop writing…', 'woocommerce' ) }
+			</button>
 		);
-	};
-
-	const generateDescription = async () => {
-		try {
-			setFetching( true );
-			const response = await apiFetch< WooApiResponse >( {
-				path: '/wooai/text-generation',
-				method: 'POST',
-				data: {
-					prompt: buildPrompt(),
-				},
-			} );
-			tinyEditor.setContent( response.generated_text || '' );
-		} catch ( e ) {
-			if ( ! ( e as WPAPIError )?.data?.status ) {
-				tinyEditor.setContent( getApiError() );
-			}
-
-			tinyEditor.setContent(
-				getApiError( ( e as WPAPIError ).data.status )
-			);
-			/* eslint-disable no-console */
-			console.error( e );
-		}
-		setFetching( false );
-	};
+	}
 
 	return (
 		<button
```

Comment:
Same as above, should we put these two buttons into their own components?
IMO it can be faster to skim this file if it looks something like:

```
	{
		completionActive ? (
			<StopCompletionButton {props} />
		) : (
			<GenerateCompletionButton  {props}/>
		)
	}
}
```
I know it's not fun passing down multiple props (although one of these is a hook which makes it easier) so I'll defer to your React expertise on whether separating them out into components for readability is worth it.

Reply:
I was a little on the fence about this, but gave it a go in https://github.com/woocommerce/woocommerce/pull/38407/commits/20b37d365f2618c96ea84d125a09c51204af7f5a. I think it's a bit cleaner, and I just left a couple props (onClick/disabled) in the root file to prevent too much prop drilling. 👍🏻